core/src/main/java/org/zenoss/zep/dao/impl/EventSummaryDaoImpl.java

```diff
@@ -179,167 +177,314 @@ public String create(Event event, EventPreCreateContext context) throws ZepExcep
+        fields.put(COLUMN_EVENT_COUNT, summary.getCount());
 
-        long updateTime = System.currentTimeMillis();
-        String indexSql = "INSERT INTO event_summary_index_queue (uuid, update_time) " +
-                    "SELECT uuid, " + String.valueOf(updateTime) + " " +
-                    "FROM event_summary " + 
-                    "WHERE fingerprint_hash=:fingerprint_hash";
-        this.template.update(indexSql, updateFields);
+        final String createdUuid = this.uuidGenerator.generate().toString();
+        fields.put(COLUMN_UUID, uuidConverter.toDatabaseType(createdUuid));
+        if (createClearHash) {
+            fields.put(COLUMN_CLEAR_FINGERPRINT_HASH, EventDaoUtils.createClearHash(event,
+                    context.getClearFingerprintGenerator()));
+        }
 
-        this.template.update(updateSql.toString(), updateFields);
-        return oldSummary.getUuid();
+        return fields;
     }
 
-    /**
-     * When an event is de-duped, if the event occurrence has a created time greater than or equal to the current
-     * last_seen for the event summary, these fields from the event summary row are overwritten by values from the new
-     * event occurrence. Special handling is performed when de-duping for event status and event details.
-     */
-    private static final List<String> UPDATE_FIELD_NAMES = Arrays.asList(COLUMN_EVENT_GROUP_ID,
-            COLUMN_EVENT_CLASS_ID, COLUMN_EVENT_CLASS_KEY_ID, COLUMN_EVENT_CLASS_MAPPING_UUID, COLUMN_EVENT_KEY_ID,
-            COLUMN_SEVERITY_ID, COLUMN_ELEMENT_UUID, COLUMN_ELEMENT_TYPE_ID, COLUMN_ELEMENT_IDENTIFIER,
-            COLUMN_ELEMENT_TITLE, COLUMN_ELEMENT_SUB_UUID, COLUMN_ELEMENT_SUB_TYPE_ID, COLUMN_ELEMENT_SUB_IDENTIFIER,
-            COLUMN_ELEMENT_SUB_TITLE, COLUMN_LAST_SEEN, COLUMN_MONITOR_ID, COLUMN_AGENT_ID, COLUMN_SYSLOG_FACILITY,
-            COLUMN_SYSLOG_PRIORITY, COLUMN_NT_EVENT_CODE, COLUMN_CLEAR_FINGERPRINT_HASH, COLUMN_SUMMARY, COLUMN_MESSAGE,
-            COLUMN_TAGS_JSON);
-
-    private Map<String,Object> getUpdateFields(EventSummaryOrBuilder oldSummary, Event occurrence,
-                                                Map<String,Object> insertFields) throws ZepException {
-        TypeConverter<Long> timestampConverter = databaseCompatibility.getTimestampConverter();
-        Map<String,Object> updateFields = new HashMap<String, Object>();
-
-        // Always increment count
-        updateFields.put(COLUMN_EVENT_COUNT, oldSummary.getCount() + occurrence.getCount());
-
-        updateFields.put(COLUMN_UPDATE_TIME, insertFields.get(COLUMN_UPDATE_TIME));
+    private String saveEventByFingerprint(final byte[] fingerprintHash, final Collection<Event> events,
+                                          final EventPreCreateContext context, final boolean createClearHash)
+            throws ZepException
+    {
+        try {
+            return metricRegistry.timer(getClass().getName() + ".saveEventByFingerprint").time(new Callable<String>() {
```

Comment:
@Timed only works on public methods. Regardless, while I was rough coding, I did make it public so I could quickly time it. It mysteriously didn't work on this method. I don't know why.